Sources/EmbraceConfigInternal/EmbraceConfig.swift

```diff
@@ -54,21 +57,24 @@ public class EmbraceConfig {
         }
 
         update()
-        lastUpdateTime = Date().timeIntervalSince1970
         return true
     }
 
     public func update() {
-        configurable.update { [weak self] didChange, error in
-            if let error = error {
-                self?.logger.error(
-                    "Failed update in EmbraceConfig",
-                    attributes: [ "error.message": error.localizedDescription ]
-                )
-            }
-
-            if didChange {
-                self?.notificationCenter.post(name: .embraceConfigUpdated, object: self)
+        self.queue.async { [weak self] in
+            self?.configurable.update { [weak self] didChange, error in
+                if let error = error {
+                    self?.logger.error(
+                        "Failed update in EmbraceConfig",
+                        attributes: [ "error.message": error.localizedDescription ]
+                    )
+                }
+
+                self?.lastUpdateTime = Date().timeIntervalSince1970
```

Comment:
If `updateIfNeeded` is called for the first time many times in quick succession its possible that we kick off multiple jobs onto this async queue.

Should we guard against this with the `lastUpdateTime` or should we introduce a variable to track that an update is currently in progress?